mindmeld/components/_elasticsearch_helpers.py

```diff
@@ -272,26 +341,28 @@ def load_index(
         # create the progess bar with docs count
         pbar = tqdm(total=docs_count)
 
-        for okay, result in streaming_bulk(
-            es_client,
-            docs,
-            index=scoped_index_name,
-            doc_type=doc_type,
-            chunk_size=50,
-            raise_on_error=False,
+        es_version_7 = is_es_version_7(es_client)
+        for okay, result in version_compatible_streaming_bulk(
+            es_client, docs, scoped_index_name, 50, False, DOC_TYPE
         ):
-
             action, result = result.popitem()
-            doc_id = "/%s/%s/%s" % (index_name, doc_type, result["_id"])
+            if es_version_7:
+                doc_id = "/%s/%s" % (index_name, result["_id"])
+            else:
+                doc_id = "/%s/%s/%s" % (index_name, doc_type, result["_id"])
+
             # process the information from ES whether the document has been
             # successfully indexed
             if not okay:
                 logger.error("Failed to %s document %s: %r", action, doc_id, result)
             else:
                 count += 1
             pbar.update(1)
+
         # close the progress bar and flush all output
         pbar.close()
+        es_client.indices.flush(index=scoped_index_name)
+        es_client.indices.refresh(index=scoped_index_name)
```

Comment:
Could you add some documentation here on why we do this?

Reply:
Flushing makes sure data is permanently stored in the Lucene index: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-flush.html#flush-api-all-ex
Refreshing makes all operations performed on the index available for search: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html

Can add to the code as well.